src/cluster.h

```diff
@@ -209,11 +212,13 @@ typedef struct {
     uint32_t ping_sent;
     uint32_t pong_received;
     char ip[NET_IP_STR_LEN];  /* IP address last time it was seen */
+    char human_readable_name[CLUSTER_HUMAN_NAMELEN]; /* Human readable name for node */
     uint16_t port;              /* base port last time it was seen */
     uint16_t cport;             /* cluster port last time it was seen */
     uint16_t flags;             /* node->flags copy */
     uint16_t pport;             /* plaintext-port, when base port is TLS */
     uint16_t notused1;
+    int has_human_readable_name; /* Indicates whether custom nodename is used */
```

Comment:
During upgrades new nodes won't be able to exchange messages with old nodes, since the layout of data has changed. Take a look at this PR, https://github.com/redis/redis/pull/9530, which adds a way to add extensions (it's not merged but hopefull will be soon). I also don't think we need to gossip the hostname, as nodes will eventually learn the hostname through regular ping/pong messages.